From a75e5161a65e71925795f1161e251fd297e41cb0 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Mon, 27 Mar 2023 22:04:40 +0200
Subject: [PATCH 46/57] Revert "maple_tree: remove extra smp_wmb() from
 mas_dead_leaves()"

This reverts commit 59f5137e9dc1eb30a2f06fcadb75a8b21ca7e117.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 lib/maple_tree.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/maple_tree.c b/lib/maple_tree.c
index 96d673e4b..946acda29 100644
--- a/lib/maple_tree.c
+++ b/lib/maple_tree.c
@@ -5503,6 +5503,7 @@ unsigned char mas_dead_leaves(struct ma_state *mas, void __rcu **slots,
 			break;
 
 		mte_set_node_dead(entry);
+		smp_wmb(); /* Needed for RCU */
 		node->type = type;
 		rcu_assign_pointer(slots[offset], node);
 	}
-- 
2.40.0.71.g950264636c

